pkgm.ts

```diff
@@ -422,3 +430,119 @@ function get_pkgx() {
   }
   throw new Error("no `pkgx` found in `$PATH`");
 }
+
+async function* ls() {
+  for (
+    const path of [new Path("/usr/local/pkgs"), Path.home().join(".local/pkgs")]
+  ) {
+    if (!path.isDirectory()) continue;
+    const dirs = [path];
+    let dir: Path | undefined;
+    while ((dir = dirs.pop()) != undefined) {
+      for await (const [path, { name, isDirectory, isSymlink }] of dir.ls()) {
+        if (!isDirectory || isSymlink) continue;
+        if (/^v\d+\./.test(name)) {
```

Comment:
should we test for `${basePath}/pkgs` as well, to be careful?

Reply:
shorthand. what i meant was, since _anything_ could be here, testing to ensure the link looks like `basePath`/pkgs/whatever/vFOO/ is safer.

but that's probably a v1-grade safety feature too.